### ISSUE TITLE
Dynamic Withdraw: Broken for Blackout

### ISSUE THREAD

**Issue by CWolfs** [CWolfs/MissionControl#314]
Dynamic Withdraw seems to be broken for `Blackout` contract type.

Investigate and fix.

**Comment by CWolfs** [comment]
```
2020-05-18T17:11:14 - Log - MissionControl [DEBUG] [CombatHUDRetreatEscMenuOnRetreatConfirmedPatch] Sending Dynamic Withdraw Trigger Message
2020-05-18T17:11:14 - Log - MissionControl [DEBUG] [ChunkMatchesChunkGuidConditional] Evaluating...
2020-05-18T17:11:14 - Log - MissionControl [DEBUG] [PositionRegion] Positioning Region...
2020-05-18T17:11:14 - Log - MissionControl [DEBUG] [PositionRegion] Finding possible position...
2020-05-18T17:11:14 - Warning - MissionControl [WARNING] [IsWithinBoundedDistanceOfTarget] Distance is 410.1122 and so not within bounds. Getting new random position
2020-05-18T17:11:14 - Warning - MissionControl [WARNING] [GetRandomPositionFromTarget] Position is not within bounds. Getting new random position
2020-05-18T17:11:14 - Log - MissionControl [DEBUG] 
2020-05-18T17:11:14 - Log - MissionControl [DEBUG] -------- [PFM.IsSpawnValid] [PositionRegionResult.Region_Withdraw] --------
2020-05-18T17:11:14 - Log - MissionControl [DEBUG] [PFM.IsSpawnValid] [PositionRegionResult.Region_Withdraw] Spawn point of '41.11526' is too steep (> 39). Not a valid spawn
2020-05-18T17:11:14 - Log - MissionControl [DEBUG] [PositionRegion] Trying again to find a possible position...
2020-05-18T17:11:14 - Log - MissionControl [DEBUG] 
2020-05-18T17:11:14 - Log - MissionControl [DEBUG] -------- [PFM.IsSpawnValid] [PositionRegionResult.Region_Withdraw] --------
2020-05-18T17:11:14 - Log - MissionControl [DEBUG] [PFM.IsSpawnValid] [PositionRegionResult.Region_Withdraw] Path count is: '16', Current position is: '(-456.0, 260.4, 207.8)'
2020-05-18T17:11:14 - Log - MissionControl [DEBUG] [PFM.IsSpawnValid] [PositionRegionResult.Region_Withdraw] Path count is: '16', Current position is: '(-456.0, 260.4, 207.8)'
2020-05-18T17:11:14 - Log - MissionControl [DEBUG] [PFM.IsSpawnValid] [PositionRegionResult.Region_Withdraw] Last point is '(-492.0, 0.0, -103.9)', Validity position is '(-480.0, 177.7, -124.7)'
2020-05-18T17:11:14 - Log - MissionControl [DEBUG] [PFM.IsSpawnValid] [PositionRegionResult.Region_Withdraw] Distance from last path to valdity position is: '24' and is it within zone radius? 'True'
2020-05-18T17:11:14 - Log - MissionControl [DEBUG] [PFM.IsSpawnValid] [PositionRegionResult.Region_Withdraw] Has valid long range path finding
2020-05-18T17:11:14 - Log - MissionControl [DEBUG] [PFM.HasValidNeighbours] Cost of neighbour is 11.43153 with max pathfinder cost being 210
2020-05-18T17:11:14 - Log - MissionControl [DEBUG] [PFM.HasValidNeighbours] Cost of neighbour is 17.46196 with max pathfinder cost being 210
2020-05-18T17:11:14 - Log - MissionControl [DEBUG] [PFM.IsSpawnValid] [PositionRegionResult.Region_Withdraw] Has at least two valid neighbours
2020-05-18T17:11:14 - Error - MessageCenter [ERROR] CRITICAL ERROR, PLEASE REPORT:
Delegate OnMessage - Standard for message type 10000 failed with exception 
Object reference not set to an instance of an object
  at BattleTech.AbstractActor.IsFriendly (BattleTech.ICombatant target) [0x0000c] in <4184af8dbeb44635831353f4d349631c>:0 
  at BattleTech.PathNodeGrid.BuildPathNetwork (System.Int32 numThisFrame) [0x00164] in <4184af8dbeb44635831353f4d349631c>:0 
  at BattleTech.PathNodeGrid.UpdateBuild (System.Int32 numThisFrame) [0x00008] in <4184af8dbeb44635831353f4d349631c>:0 
  at BattleTech.Pathing.UpdateBuild (System.Int32 numBuiltThisFrame) [0x00006] in <4184af8dbeb44635831353f4d349631c>:0 
  at MissionControl.PathFinderManager.HasValidLocalPathfinding (BattleTech.PathNode positionNode, UnityEngine.Vector3 validityPosition, BattleTech.UnitType type) [0x0004f] in <f5154dd0af2847149be65305b8d8f66e>:0 
  at MissionControl.PathFinderManager.IsSpawnValid (UnityEngine.GameObject spawnGo, UnityEngine.Vector3 position, UnityEngine.Vector3 validityPosition, BattleTech.UnitType type, System.String identifier) [0x003d4] in <f5154dd0af2847149be65305b8d8f66e>:0 
  at MissionControl.Result.PositionRegionResult.Trigger (MessageCenterMessage inMessage, System.String triggeringName) [0x000f2] in <f5154dd0af2847149be65305b8d8f66e>:0 
  at BattleTech.Framework.TriggerResponse.Transmit (MessageCenterMessage message, BattleTech.Framework.SignalType originalSignalType, System.String transmittingName) [0x00069] in <4184af8dbeb44635831353f4d349631c>:0 
  at BattleTech.Framework.SmartTriggerResponse.OnMessage (MessageCenterMessage message) [0x0008c] in <4184af8dbeb44635831353f4d349631c>:0 
  at MessageCenter.SendMessagesForType (MessageCenterMessageType messageType, MessageCenterMessage message) [0x00037] in <4184af8dbeb44635831353f4d349631c>:0 
2020-05-18T17:11:14 - Log - MissionControl [LOG] [DialogResultPatch Prefix] Patching Trigger
2020-05-18T17:11:14 - Log - CombatHud.OnTriggerDialog
2020-05-18T17:11:15 - Exception - NullReferenceException: Object reference not set to an instance of an object
2020-05-18T17:11:15 - BattleTech.AbstractActor.IsFriendly (BattleTech.ICombatant target) (at <4184af8dbeb44635831353f4d349631c>:0)
BattleTech.PathNodeGrid.BuildPathNetwork (System.Int32 numThisFrame) (at <4184af8dbeb44635831353f4d349631c>:0)
BattleTech.PathNodeGrid.UpdateBuild (System.Int32 numThisFrame) (at <4184af8dbeb44635831353f4d349631c>:0)
BattleTech.Pathing.UpdateBuild (System.Int32 numBuiltThisFrame) (at <4184af8dbeb44635831353f4d349631c>:0)
BattleTech.PathingManager.Update () (at <4184af8dbeb44635831353f4d349631c>:0)
BattleTech.CombatGameState.Update () (at <4184af8dbeb44635831353f4d349631c>:0)
BattleTech.GameInstance.Update (System.Single deltaTime) (at <4184af8dbeb44635831353f4d349631c>:0)
(wrapper dynamic-method) BattleTech.UnityGameInstance.Update_Patch1(object)
```

**Comment by CWolfs** [comment]
This is similar to problems reported by the RT Team for v0.4.6. I put in a fix for this in v1.0.0 but it didn't work. Putting in a new fix.

**Comment by CWolfs** [comment]
In `develop`.
